### ISSUE TITLE
Failed to build rtags on CentOS7 machine

### ISSUE THREAD

**Issue by haji-ali** [Andersbakken/rtags#1292]
I cloned rtags on my CentOS 7 machine and tried building it but got the following output. It seems that some features are missing (`mach_absolute_time`, `kqueue`, `SO_NOSIGPIPE` and `GetLogicalProcessorInformation`), even though I installed all the requirement for rtags.
Is there a requirement I am missing?

```
-- The C compiler identification is GNU 7.3.1
-- The CXX compiler identification is GNU 7.3.1
-- Check for working C compiler: /opt/rh/devtoolset-7/root/usr/bin/cc
-- Check for working C compiler: /opt/rh/devtoolset-7/root/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /opt/rh/devtoolset-7/root/usr/bin/c++
-- Check for working CXX compiler: /opt/rh/devtoolset-7/root/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- llvm-config executable found: /usr/bin/llvm-config
usage: llvm-config <OPTION>... [<COMPONENT>...]

Get various configuration information needed to compile programs which use
LLVM.  Typically called from 'configure' scripts.  Examples:
  llvm-config --cxxflags
  llvm-config --ldflags
  llvm-config --libs engine bcreader scalaropts

Options:
  --version         Print LLVM version.
  --prefix          Print the installation prefix.
  --src-root        Print the source root LLVM was built from.
  --obj-root        Print the object root used to build LLVM.
  --bindir          Directory containing LLVM executables.
  --includedir      Directory containing LLVM headers.
  --libdir          Directory containing LLVM libraries.
  --cppflags        C preprocessor flags for files that include LLVM headers.
  --cflags          C compiler flags for files that include LLVM headers.
  --cxxflags        C++ compiler flags for files that include LLVM headers.
  --ldflags         Print Linker flags.
  --libs            Libraries needed to link against LLVM components.
  --libnames        Bare library names for in-tree builds.
  --libfiles        Fully qualified library filenames for makefile depends.
  --components      List of all possible components.
  --targets-built   List of all targets currently built.
  --host-target     Target triple used to configure LLVM.
  --build-mode      Print build mode of LLVM tree (e.g. Debug or Release).
Typical components:
  all               All LLVM libraries (default).
  engine            Either a native JIT or a bitcode interpreter.
-- Using Clang version 3.4.2 from /usr/lib64/llvm with CXXFLAGS -I/usr/include -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D_FORTIFY_SOURCE
-- Found LibClang: /usr/lib64/llvm/libclang.so  
-- Performing Test LIBCLANG_COMPILES
-- Performing Test LIBCLANG_COMPILES - Success
-- Performing Test CXX_11_EXTENDED_REGEX_SUPPORT
-- Performing Test CXX_11_EXTENDED_REGEX_SUPPORT - Success
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Looking for backtrace
-- Looking for backtrace - found
-- Looking for CLOCK_MONOTONIC_RAW
-- Looking for CLOCK_MONOTONIC_RAW - found
-- Looking for CLOCK_MONOTONIC
-- Looking for CLOCK_MONOTONIC - found
-- Looking for mach_absolute_time
-- Looking for mach_absolute_time - not found
-- Looking for inotify_init
-- Looking for inotify_init - found
-- Looking for kqueue
-- Looking for kqueue - not found
-- Looking for epoll_wait
-- Looking for epoll_wait - found
-- Looking for select
-- Looking for select - found
-- Looking for FD_CLOEXEC
-- Looking for FD_CLOEXEC - found
-- Looking for SO_NOSIGPIPE
-- Looking for SO_NOSIGPIPE - not found
-- Looking for MSG_NOSIGNAL
-- Looking for MSG_NOSIGNAL - found
-- Looking for GetLogicalProcessorInformation
-- Looking for GetLogicalProcessorInformation - not found
-- Looking for SCHED_IDLE
-- Looking for SCHED_IDLE - found
-- Looking for SHM_DEST
-- Looking for SHM_DEST - found
-- Performing Test HAVE_STATMTIM
-- Performing Test HAVE_STATMTIM - Success
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.7") 
CMake Error at /usr/share/cmake/Modules/FindOpenSSL.cmake:278 (list):
  list GET given empty list
Call Stack (most recent call first):
  src/rct/rct.cmake:96 (find_package)
  src/CMakeLists.txt:136 (include)


CMake Error at /usr/share/cmake/Modules/FindOpenSSL.cmake:279 (list):
  list GET given empty list
Call Stack (most recent call first):
  src/rct/rct.cmake:96 (find_package)
  src/CMakeLists.txt:136 (include)


CMake Error at /usr/share/cmake/Modules/FindOpenSSL.cmake:281 (list):
  list GET given empty list
Call Stack (most recent call first):
  src/rct/rct.cmake:96 (find_package)
  src/CMakeLists.txt:136 (include)


CMake Error at /usr/share/cmake/Modules/FindOpenSSL.cmake:283 (list):
  list GET given empty list
Call Stack (most recent call first):
  src/rct/rct.cmake:96 (find_package)
  src/CMakeLists.txt:136 (include)


-- Found OpenSSL: /usr/lib64/libssl.so;/usr/lib64/libcrypto.so (found version ".0.0`") 
-- Performing Test HAVE_CXX11
-- Performing Test HAVE_CXX11 - Success
-- Performing Test HAVE_STRING_ITERATOR_ERASE
-- Performing Test HAVE_STRING_ITERATOR_ERASE - Success
-- Found emacs version 25.3.1
-- 
-- The following OPTIONAL packages have been found:

 * ZLIB
 * OpenSSL
 * PkgConfig , helper tool , <https://www.freedesktop.org/wiki/Software/pkg-config/>
   We use it to get the bash completion installation path, and replace the prefix with the value of CMAKE_INSTALL_PREFIX.
 * Perl

-- The following REQUIRED packages have been found:

 * LibClang

-- Configuring incomplete, errors occurred!
See also "rtags/CMakeFiles/CMakeOutput.log".
See also "rtags/CMakeFiles/CMakeError.log".
```  

CMakeError.log has the following:
```
Determining if the mach_absolute_time exist failed with the following output:
Change Dir: /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp

Run Build Command:/usr/bin/gmake "cmTryCompileExec29815525/fast"
/usr/bin/gmake -f CMakeFiles/cmTryCompileExec29815525.dir/build.make CMakeFiles/cmTryCompileExec29815525.dir/build
gmake[1]: Entering directory `/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp'
/usr/bin/cmake -E cmake_progress_report /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CMakeFiles 1
Building CXX object CMakeFiles/cmTryCompileExec29815525.dir/CheckSymbolExists.cxx.o
/opt/rh/devtoolset-7/root/usr/bin/c++    -I/usr/include -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D_FORTIFY_SOURCE -Wall -Wextra -Wpointer-arith -Wnon-virtual-dtor -Wformat -Wshadow -std=c++11 -pthread    -o CMakeFiles/cmTryCompileExec29815525.dir/CheckSymbolExists.cxx.o -c /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx
/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:2:10: fatal error: mach/mach.h: No such file or directory
 #include <mach/mach.h>
          ^~~~~~~~~~~~~
compilation terminated.
gmake[1]: *** [CMakeFiles/cmTryCompileExec29815525.dir/CheckSymbolExists.cxx.o] Error 1
gmake[1]: Leaving directory `/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp'
gmake: *** [cmTryCompileExec29815525/fast] Error 2

File /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:
/* */
#include <mach/mach.h>
#include <mach/mach_time.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef mach_absolute_time
  return ((int*)(&mach_absolute_time))[argc];
#else
  (void)argc;
  return 0;
#endif
}

Determining if the kqueue exist failed with the following output:
Change Dir: /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp

Run Build Command:/usr/bin/gmake "cmTryCompileExec3686170203/fast"
/usr/bin/gmake -f CMakeFiles/cmTryCompileExec3686170203.dir/build.make CMakeFiles/cmTryCompileExec3686170203.dir/build
gmake[1]: Entering directory `/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp'
/usr/bin/cmake -E cmake_progress_report /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CMakeFiles 1
Building CXX object CMakeFiles/cmTryCompileExec3686170203.dir/CheckSymbolExists.cxx.o
/opt/rh/devtoolset-7/root/usr/bin/c++    -I/usr/include -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D_FORTIFY_SOURCE -Wall -Wextra -Wpointer-arith -Wnon-virtual-dtor -Wformat -Wshadow -std=c++11 -pthread    -o CMakeFiles/cmTryCompileExec3686170203.dir/CheckSymbolExists.cxx.o -c /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx
In file included from /usr/include/sys/types.h:25:0,
                 from /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:2:
/usr/include/features.h:330:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
 #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
    ^~~~~~~
/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:3:10: fatal error: sys/event.h: No such file or directory
 #include <sys/event.h>
          ^~~~~~~~~~~~~
compilation terminated.
gmake[1]: *** [CMakeFiles/cmTryCompileExec3686170203.dir/CheckSymbolExists.cxx.o] Error 1
gmake[1]: Leaving directory `/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp'
gmake: *** [cmTryCompileExec3686170203/fast] Error 2

File /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:
/* */
#include <sys/types.h>
#include <sys/event.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef kqueue
  return ((int*)(&kqueue))[argc];
#else
  (void)argc;
  return 0;
#endif
}

Determining if the SO_NOSIGPIPE exist failed with the following output:
Change Dir: /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp

Run Build Command:/usr/bin/gmake "cmTryCompileExec3665370734/fast"
/usr/bin/gmake -f CMakeFiles/cmTryCompileExec3665370734.dir/build.make CMakeFiles/cmTryCompileExec3665370734.dir/build
gmake[1]: Entering directory `/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp'
/usr/bin/cmake -E cmake_progress_report /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CMakeFiles 1
Building CXX object CMakeFiles/cmTryCompileExec3665370734.dir/CheckSymbolExists.cxx.o
/opt/rh/devtoolset-7/root/usr/bin/c++    -I/usr/include -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D_FORTIFY_SOURCE -Wall -Wextra -Wpointer-arith -Wnon-virtual-dtor -Wformat -Wshadow -std=c++11 -pthread    -o CMakeFiles/cmTryCompileExec3665370734.dir/CheckSymbolExists.cxx.o -c /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx
In file included from /usr/include/sys/types.h:25:0,
                 from /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:2:
/usr/include/features.h:330:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
 #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
    ^~~~~~~
/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx: In function ‘int main(int, char**)’:
/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:9:19: error: ‘SO_NOSIGPIPE’ was not declared in this scope
   return ((int*)(&SO_NOSIGPIPE))[argc];
                   ^~~~~~~~~~~~
/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:9:19: note: suggested alternative: ‘SO_OOBINLINE’
   return ((int*)(&SO_NOSIGPIPE))[argc];
                   ^~~~~~~~~~~~
                   SO_OOBINLINE
gmake[1]: *** [CMakeFiles/cmTryCompileExec3665370734.dir/CheckSymbolExists.cxx.o] Error 1
gmake[1]: Leaving directory `/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp'
gmake: *** [cmTryCompileExec3665370734/fast] Error 2

File /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:
/* */
#include <sys/types.h>
#include <sys/socket.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef SO_NOSIGPIPE
  return ((int*)(&SO_NOSIGPIPE))[argc];
#else
  (void)argc;
  return 0;
#endif
}

Determining if the GetLogicalProcessorInformation exist failed with the following output:
Change Dir: /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp

Run Build Command:/usr/bin/gmake "cmTryCompileExec2636387088/fast"
/usr/bin/gmake -f CMakeFiles/cmTryCompileExec2636387088.dir/build.make CMakeFiles/cmTryCompileExec2636387088.dir/build
gmake[1]: Entering directory `/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp'
/usr/bin/cmake -E cmake_progress_report /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CMakeFiles 1
Building CXX object CMakeFiles/cmTryCompileExec2636387088.dir/CheckSymbolExists.cxx.o
/opt/rh/devtoolset-7/root/usr/bin/c++    -I/usr/include -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D_FORTIFY_SOURCE -Wall -Wextra -Wpointer-arith -Wnon-virtual-dtor -Wformat -Wshadow -std=c++11 -pthread    -o CMakeFiles/cmTryCompileExec2636387088.dir/CheckSymbolExists.cxx.o -c /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx
/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:2:10: fatal error: windows.h: No such file or directory
 #include <windows.h>
          ^~~~~~~~~~~
compilation terminated.
gmake[1]: *** [CMakeFiles/cmTryCompileExec2636387088.dir/CheckSymbolExists.cxx.o] Error 1
gmake[1]: Leaving directory `/home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp'
gmake: *** [cmTryCompileExec2636387088/fast] Error 2

File /home/ah180/Work/Apps/rtags/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:
/* */
#include <windows.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef GetLogicalProcessorInformation
  return ((int*)(&GetLogicalProcessorInformation))[argc];
#else
  (void)argc;
  return 0;
#endif
}
```

**Comment by cslux** [comment]
Hmm, maybe the kernel development header files are missing or glibc devel, so
not 100 % sure about it.

**Comment by cslux** [comment]
@haji-ali That's really weird. I successfully built rtags in a VirtualMachine running CentOS 7 from osboxes.org yesterday. Not quite sure whats going on here. What are your build steps?

**Comment by cslux** [comment]
@haji-ali Whats your cmake version?

**Comment by haji-ali** [comment]
cmake is 2.8.12.2

I just cloned the repo recursively and called
```
LIBCLANG_LLVM_CONFIG_EXECUTABLE=/usr/bin/llvm-config cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=1 .
```

I had to specify the config path because otherwise cmake would pick some other wrong version on my machine. I have to say that this is my work computer so I am not sure if there is some incompatibility with a library that someone else installed. I am just really not sure how to troubleshoot this.

I noticed that others had this problem here (See #486 and #1149)
